src/main/java/org/terasology/launcher/TerasologyLauncher.java

```diff
@@ -16,111 +16,86 @@
-import javafx.event.EventHandler;
 import javafx.fxml.FXMLLoader;
-import javafx.geometry.Pos;
-import javafx.geometry.Rectangle2D;
 import javafx.scene.Parent;
 import javafx.scene.Scene;
-import javafx.scene.control.Label;
-import javafx.scene.control.ProgressBar;
-import javafx.scene.effect.DropShadow;
 import javafx.scene.image.Image;
-import javafx.scene.image.ImageView;
-import javafx.scene.layout.Pane;
-import javafx.scene.layout.VBox;
-import javafx.stage.Screen;
 import javafx.stage.Stage;
-import javafx.stage.StageStyle;
-import javafx.util.Duration;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.terasology.crashreporter.CrashReporter;
-import org.terasology.launcher.gui.javafx.ApplicationController;
+import org.terasology.launcher.gui.javafx.MainController;
 import org.terasology.launcher.log.TempLogFilePropertyDefiner;
 import org.terasology.launcher.util.BundleUtils;
 import org.terasology.launcher.util.Languages;
-import org.terasology.launcher.util.LauncherStartFailedException;
 import org.terasology.launcher.version.TerasologyLauncherVersionInfo;
 
 import java.io.IOException;
 import java.nio.file.Path;
 import java.util.Arrays;
 import java.util.List;
 import java.util.MissingResourceException;
-import java.util.ResourceBundle;
 
 public final class TerasologyLauncher extends Application {
 
     private static final Logger logger = LoggerFactory.getLogger(TerasologyLauncher.class);
 
-    private static final int SPLASH_WIDTH = 800;
-    private static final int SPLASH_HEIGHT = 223;
-
-    private Pane splashLayout;
-    private ProgressBar loadProgress;
-    private Label progressText;
-    private Stage mainStage;
-    private HostServices hostServices;
+    private final LauncherInitTask initTask = new LauncherInitTask();
+    private LauncherConfiguration initialConfig;
+    private MainController mainController;
 
     public static void main(String[] args) {
         launch(args);
     }
 
     @Override
-    public void init() throws Exception {
-        ImageView splash = new ImageView(BundleUtils.getFxImage("splash"));
-        loadProgress = new ProgressBar();
-        loadProgress.setPrefWidth(SPLASH_WIDTH);
-        progressText = new Label();
-        splashLayout = new VBox();
-        splashLayout.getChildren().addAll(splash, loadProgress, progressText);
-        progressText.setAlignment(Pos.CENTER);
-        splashLayout.getStylesheets().add(BundleUtils.getStylesheet("css_splash"));
-        splashLayout.setEffect(new DropShadow());
-        hostServices = getHostServices();
+    public void init() {
+        initTask.setOnSucceeded(event -> {
+            if (mainController != null) {
+                mainController.update(initTask.getValue());
+            } else {
+                initialConfig = initTask.getValue();
+            }
+        });
+        initTask.setOnFailed(event -> openCrashReporterAndExit((Exception) initTask.getException()));
```

Comment:
Please don't cast throwable to exception. Throwable is more general than exception, so this is a down-cast. Up-casts, e.g. treating an oak tree as a tree in general, are fine as we can be sure that any oak always also is a tree. The other way round however does not hold, as we cannot be sure, that every tree is an oak - it could also be a palm tree.

Afair the crash reporter takes a general throwable as an argument, so you should change the signature of `openCrashReporterAndExit` to accept a throwable, too.